src/store/reducers/movies.js

```diff
@@ -0,0 +1,97 @@
+import { SET_MOVIES, SET_IS_LOADING, DELETE_MOVIE, SORT_MOVIES, FILTER_MOVIES, ADD_MOVIE, EDIT_MOVIE } from 'store/actionTypes';
+import sortOrder from 'enums/sortOrder';
+import filterTerms from 'enums/filterTerms';
+import getRandomNum from 'utils/getRandomNum';
+
+const initialState = {
+  movies: [],
+  isLoading: false
+}
+
+const sortMap = {
+  [sortOrder.title]: 'title',
+  [sortOrder.year]: 'label',
+  [sortOrder.rating]: 'rating'
+}
+
+let allMovies = [];
+
+const updateAllMovies = (movies) => {
+  allMovies = movies;
+}
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case FILTER_MOVIES: {
+      return {
+        ...state,
+        movies:
+          action.payload.filterTerm === filterTerms.all ?
+            allMovies :
+            allMovies.filter(({ subtitle }) => subtitle.includes(action.payload.filterTerm))
```

Comment:
I think task requires to do filtration/sorting through api request
`/movies?search=${query}&searchBy=title&sortBy=${sortedBy}&sortOrder=desc&filter=${filteredBy}`

Reply:
Are you sure? I haven't seen such requirements in task description. Also there's nothing about it in swagger docs